modules/apps/transfer/ibc_module_test.go

```diff
@@ -273,6 +276,157 @@ func (suite *TransferTestSuite) TestOnChanOpenAck() {
+			},
+			false,
+			false,
+			"sender address cannot be blank: invalid address: invalid type",
+		},
+	}
+
+	for _, tc := range testCases {
+		suite.Run(tc.name, func() {
+			suite.SetupTest() // reset
+
+			path := ibctesting.NewTransferPath(suite.chainA, suite.chainB)
+			path.Setup()
+
+			packetData := types.NewFungibleTokenPacketDataV2(
+				[]types.Token{
+					{
+						Denom:  types.NewDenom(sdk.DefaultBondDenom),
+						Amount: sdkmath.NewInt(100).String(),
+					},
+				},
+				suite.chainA.SenderAccount.GetAddress().String(),
+				suite.chainB.SenderAccount.GetAddress().String(),
+				"",
+				nil,
+			)
+
+			seq := uint64(1)
+			packet = channeltypes.NewPacket(packetData.GetBytes(), seq, path.EndpointA.ChannelConfig.PortID, path.EndpointA.ChannelID, path.EndpointB.ChannelConfig.PortID, path.EndpointB.ChannelID, clienttypes.NewHeight(1, 100), 0)
+
+			module, _, err := suite.chainA.App.GetIBCKeeper().PortKeeper.LookupModuleByPort(suite.chainA.GetContext(), ibctesting.TransferPort)
+			suite.Require().NoError(err)
+
+			cbs, ok := suite.chainA.App.GetIBCKeeper().PortKeeper.Route(module)
+			suite.Require().True(ok)
+
+			tc.malleate() // change fields in packet
+
+			ctx := suite.chainA.GetContext()
+			ack := cbs.OnRecvPacket(ctx, packet, suite.chainA.SenderAccount.GetAddress())
+
+			if tc.expAckSuccess {
+				if tc.expAsyncAck {
+					suite.Require().Nil(ack)
+				} else {
+					suite.Require().True(ack.Success())
+
+					expectedAck := channeltypes.NewResultAcknowledgement([]byte{byte(1)})
+					suite.Require().Equal(expectedAck, ack)
+				}
+
+				tokensBz, err := json.Marshal(packetData.Tokens)
+				suite.Require().NoError(err)
+
+				// In 6585 this needs to change to account for forwarded packets
+				expectedAttributes := []sdk.Attribute{
+					sdk.NewAttribute(types.AttributeKeySender, packetData.Sender),
+					sdk.NewAttribute(types.AttributeKeyReceiver, packetData.Receiver),
+					sdk.NewAttribute(types.AttributeKeyTokens, string(tokensBz)),
+					sdk.NewAttribute(types.AttributeKeyMemo, packetData.Memo),
+					sdk.NewAttribute(types.AttributeKeyAckSuccess, strconv.FormatBool(tc.expAckSuccess)),
+				}
+
+				expectedEvents := sdk.Events{
+					sdk.NewEvent(
+						types.EventTypePacket,
+						expectedAttributes...,
+					),
+				}.ToABCIEvents()
+
+				expectedEvents = sdk.MarkEventsToIndex(expectedEvents, map[string]struct{}{})
+				ibctesting.AssertEvents(&suite.Suite, expectedEvents, ctx.EventManager().Events().ToABCIEvents())
+			} else {
+				if tc.expAsyncAck {
+					suite.Require().Nil(ack)
+				} else {
+					suite.Require().False(ack.Success())
+				}
+
+				expectedAttributes := []sdk.Attribute{
+					sdk.NewAttribute(types.AttributeKeySender, ""),
+					sdk.NewAttribute(types.AttributeKeyReceiver, ""),
+					sdk.NewAttribute(types.AttributeKeyTokens, "null"),
+					sdk.NewAttribute(types.AttributeKeyMemo, ""),
+					sdk.NewAttribute(types.AttributeKeyAckSuccess, strconv.FormatBool(tc.expAckSuccess)),
```

Comment:
```suggestion
					sdk.NewAttribute(types.AttributeKeyAckSuccess, "false"),
```
maybe simpler to read?